src/FormElement/ScheduleElement/Common/FieldsUtils.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+namespace ipl\Web\FormElement\ScheduleElement\Common;
+
+use DateInterval;
+use DateTime;
+use Exception;
+use InvalidArgumentException;
+use ipl\Html\Contract\FormElement;
+use ipl\Scheduler\RRule;
+use ipl\Web\FormElement\ScheduleElement\MonthlyFields;
+
+trait FieldsUtils
+{
+    // Non-standard frequency options
+    public static $everyDay = 'day';
+    public static $everyWeekday = 'weekday';
+    public static $everyWeekend = 'weekend';
+
+    // Enumerators for the monthly and annually schedule of a custom frequency
+    public static $first = 'first';
+    public static $second = 'second';
+    public static $third = 'third';
+    public static $fourth = 'fourth';
+    public static $fifth = 'fifth';
+    public static $last = 'last';
+
+    private $regulars = [];
+
+    protected function initUtils(): void
+    {
+        $this->regulars = [
+            'MO' => $this->translate('Monday'),
+            'TU' => $this->translate('Tuesday'),
+            'WE' => $this->translate('Wednesday'),
+            'TH' => $this->translate('Thursday'),
+            'FR' => $this->translate('Friday'),
+            'SA' => $this->translate('Saturday'),
+            'SU' => $this->translate('Sunday')
+        ];
+    }
+
+    protected function createOrdinalElement(): FormElement
+    {
+        return $this->createElement('select', 'ordinal', [
+            'class'   => 'autosubmit',
+            'value'   => $this->getPopulatedValue('ordinal', static::$first),
+            'options' => [
+                static::$first  => $this->translate('First'),
+                static::$second => $this->translate('Second'),
+                static::$third  => $this->translate('Third'),
+                static::$fourth => $this->translate('Fourth'),
+                static::$fifth  => $this->translate('Fifth'),
+                static::$last   => $this->translate('Last')
+            ]
+        ]);
+    }
+
+    protected function createOrdinalSelectableDays(): FormElement
+    {
+        return $this->createElement('select', 'day', [
+            'class'   => 'autosubmit',
+            'value'   => $this->getPopulatedValue('day', static::$everyDay),
+            'options' => [
+                'Regular'       => $this->regulars,
+                'Non Standards' => [
```

Comment:
Please translate the keys.

Reply:
Or, why are they of advantage? Is it important for the user what regular/non-standard means? Or can the subsections just be removed?

Reply:
I just want to separate them. We could also do this by just rendering a line but I don't know how.

Reply:
Labels should have a purpose. If there's none or little, they should be replaced/removed. You could add an option with some dashes or a unicode line and disable it then:

` ────── ` (or longer, whatever you see fit)

Reply:
There's also a bold line char: `━━━━━━━━`